effect/shared/src/main/scala/scalaz/effect/IO.scala

```diff
@@ -161,6 +161,12 @@ sealed abstract class IO[E, A] { self =>
       case \/-(a) => IO.now[E2, A](a)
     }
 
+  /**
+   * Widens the error type to any supertype. While `leftMap` suffices for this
+   * purpose, this method is significantly faster for this purpose.
+   */
+  final def widen[E2 >: E]: IO[E2, A] = self.asInstanceOf[IO[E2, A]]
```

Comment:
no Liskov?

Reply:
If I use `<~<`, then the user will have to construct it explicitly, as these things are not automatically summoned into scope for subtyping relationships. Or am I missing something, @alexknvl?

Reply:
`implicit` will work in Scalaz8

Reply:
@alexknvl Unfortunately, I cannot use it, since the method I require (`substCv`) requires variance in the type parameter `E`.

Well, I can require it as an implicit and ignore it, performing the coercion, but then scalac will fail due to an unused implicit. 😄

Reply:
You can un-ignore something using
```scala
val _ = ev
```
But regardless, I don't feel particularly strongly about this :)

Reply:
I could go either way, too. I do feel like we need a general solution to this problem since many other classes in Scalaz 8 are invariant.

Reply:
Is there any way to create `IsCovariant[IO]` instance?

Reply:
You could have `IsCovariant` for `IO` without coercions if `IO` was an opaque type implemented by a covariant `IOImpl[+E, +A]`. 😃 (That's how `IsCovariant` is done for `Maybe`, because it is just an alias for the covariant `Option`.)